### PR TITLE
add ability to specify a specific file to write results to too

### PR DESCRIPTION
**Pull request by pes10k**
fixes brave/pagegraph-crawl#66

### CODE REVIEW

**Comment by pes10k** [comment]
i think these warnings can be safely ignored. All these uses are with arguments provided by the user, when using this as a command line too

**Comment by pes10k** [comment]
im not sure i understand, the changes i made are in validate.ts and crawl.ts.

The reason i checked in built is so that folks can use the code w/o having to download the typescript chain.  But if you'd prefer i pull it back out i can

**Comment by ShivanKaul** [comment]
Ah, do you not need any changes in run.ts? I got confused by the fact that run.js is now being built and checked in. 

Yeah, I think it would be best to not check in the built JS files.

File: src/brave/crawl.ts [hunk 1]
```diff
@@ -55,11 +57,11 @@ function createFilename (url: Url) : FilePath {
...
 }
 
-function writeToFile (args: CrawlArgs, url: Url, response: any, logger: Logger) {
-  const outputFilename = pathLib.join(
-    args.outputPath,
-    createFilename(url)
-  )
+async function writeToFile (args: CrawlArgs, url: Url, response: any, logger: Logger) {
+  const outputFilename = isDir(args.outputPath)
+    ? pathLib.join(args.outputPath, createFilename(url))
```
[the remark this answers is not in the document]
Comment (ShivanKaul) [reply src/brave/crawl.ts:62]: This is a CLI for local use, the user of the CLI provides the input.

**Comment by ShivanKaul** [comment]
Everything else LGTM, just please amend the -o help text to account for this new functionality: https://github.com/brave/pagegraph-crawl/blob/main/src/run.ts#L25

**Comment by pes10k** [comment]
run.js is being checked in bc all the *.js files are checked in, but i didnt make any changes to run.ts. and okie.

Just pushed a change that keeps built out

**Comment by pes10k** [comment]
okie pushed an update that also describes the new -o behavior

**Review by ShivanKaul — APPROVED** [review]
Looks great!

**Pull request merged**